### PR TITLE
[DEVX-1448] Add headless option for playwright

### PR DESCRIPTION
**Pull request by tianfeng92**
headless example: https://app.saucelabs.com/tests/8a7c5fdcf88a4005b5e3a349e69ea579
non-headless example: https://app.saucelabs.com/tests/ad7075ae4f1d40ec9c2b8b89797e4cb0

### CODE REVIEW

File: src/playwright-runner.js [hunk 1]
```diff
@@ -264,6 +264,7 @@ async function run (nodeBin, runCfgPath, suiteName) {
 
   process.env.PLAYWRIGHT_CFG_FILE = runCfg.playwright.configFile || '';
   process.env.BROWSER_NAME = suite.param.browserName;
+  process.env.HEADLESS = !suite.param.headed || false;
```
Comment (alexplischke) [inline src/playwright-runner.js:267]: I'd recommend sticking to the `headless` term. Neither of those places lists `headed` as an attribute:
- https://playwright.dev/docs/debug#run-in-headed-mode
- https://playwright.dev/docs/ci#running-headed
- https://playwright.dev/docs/api/class-testoptions#test-options-headless

```suggestion
  process.env.HEADLESS = suite.param.headless;
```

Comment (tianfeng92) [reply src/playwright-runner.js:267]: ok! I'll update to `headless`. I was also confused why there are two config fields `headed` and `headless`.

File: src/sauce.config.js [hunk 2]
```diff
@@ -23,7 +23,7 @@ for (const file of configFiles) {
 const overrides = {
   use: {
     browserName: process.env.BROWSER_NAME, // override browserName with suite browserName
-    headless: process.env.SAUCE_VM ? false : true,
+    headless: process.env.HEADLESS === 'true',
```
Comment (mhan83) [inline src/sauce.config.js:26]: Don't we still need to distinguish the default behaviour for docker/sauce modes? We don't want to run headed browsers in docker mode, I assume?

Comment (tianfeng92) [reply src/sauce.config.js:26]: I cannot recall why we wanted to run headless test in docker mode. And I'm not quite sure if this default setting helps. What about just setting it by config?

Comment (mhan83) [reply src/sauce.config.js:26]: I think we want headless in docker for efficiency. Assets are generated even in headless mode (I think its why we setup xvfb in the docker image?) so I don't think there's any reason to run headed in docker.

Comment (tianfeng92) [reply src/sauce.config.js]: set headless for docker as default

**Pull request merged**